x/datadeal/keeper/deal.go

```diff
@@ -574,10 +562,6 @@ func (k Keeper) DeactivateDeal(ctx sdk.Context, dealID uint64) error {
 		return sdkerrors.Wrapf(types.ErrDealDeactivate, err.Error())
 	}
 
-	if deal.Status != types.DEAL_STATUS_DEACTIVATING {
-		return sdkerrors.Wrapf(types.ErrDealDeactivate, "deal's status is not 'DEACTIVATING'")
-	}
-
```

Comment:
Remove this deal status check at `DeativateDeal`. When voting is in the deactivating state, in some cases the deal status can become `DEAL_STATUS_COMPLETED`, and  an error would occur when `DeativateDeal` is executed after a few blocks. Please check this part too.